Zend/zend_language_parser.y

```diff
@@ -979,6 +981,8 @@ expr:
 			{ $$ = zend_ast_create_binary_op(ZEND_IS_EQUAL, $1, $3); }
 	|	expr T_IS_NOT_EQUAL expr
 			{ $$ = zend_ast_create_binary_op(ZEND_IS_NOT_EQUAL, $1, $3); }
+	|	expr T_PIPE expr
+			{ $$ = zend_ast_create(ZEND_AST_CALL, $3, zend_ast_create_list(1, ZEND_AST_ARG_LIST, $1) ); $$->attr = ZEND_CALL_SYNTAX_PIPE; }
```

Comment:
The usage of pipe is flagged so that AST-manipulating tools (like code formatters) know how it was generated, and can respect that rather than accidentally reformatting it into the inside-out style.)

Reply:
You can demonstrate that by checking the flag in the `zend_ast.c` printer :)

Reply:
@Crell I think we need a test of some kind for that pipe just to make sure it really is there as expected. I've never used the zend_ast printer; what can trigger that codepath, @nikic ?

Reply:
Thanks to @iluuu1994 in chat, there's now code in zend_ast.c and a test that uses assert to re-sugar the syntax back to a pipe if appropriate.